lib/Yam/Installation/Upgrade/SelectUpdatePage.pm

```diff
@@ -0,0 +1,32 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Handle page that allows to select system/partition to be upgraded.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package Yam::Installation::Upgrade::SelectUpdatePage;
+use parent 'Installation::Navigation::NavigationBase';
+use strict;
+use warnings;
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{tbl_select_origin} = $self->{app}->table({id => "partition"});
+    return $self;
+}
+
+sub get_select_origin {
+    my ($self) = @_;
+    die "Select update page not displayed" unless $self->{tbl_select_origin}->exist();
+    return $self;
+}
+
+sub select_origin_system {
+    my ($self, $origin) = @_;
+    $self->get_select_origin->{tbl_select_origin}->select(value => $origin);
```

Comment:
don't forget for consistency (although Perl doesn't require it) to add `()` after each function.